internal/metastore/kv/kv_catalog.go

```diff
@@ -226,22 +312,73 @@ func (kc *Catalog) CreateCredential(ctx context.Context, credential *model.Crede
 	return nil
 }
 
-func (kc *Catalog) GetCollectionByID(ctx context.Context, collectionID typeutil.UniqueID, ts typeutil.Timestamp) (*model.Collection, error) {
-	collKey := fmt.Sprintf("%s/%d", CollectionMetaPrefix, collectionID)
-	collVal, err := kc.Snapshot.Load(collKey, ts)
+func (kc *Catalog) listPartitionsAfter210(ctx context.Context, collectionID typeutil.UniqueID, ts typeutil.Timestamp) ([]*model.Partition, error) {
+	prefix := buildPartitionPrefix(collectionID)
+	_, values, err := kc.Snapshot.LoadWithPrefix(prefix, ts)
 	if err != nil {
-		log.Error("get collection meta fail", zap.String("key", collKey), zap.Error(err))
 		return nil, err
 	}
+	partitions := make([]*model.Partition, 0, len(values))
+	for _, v := range values {
+		partitionMeta := &pb.PartitionInfo{}
+		err := proto.Unmarshal([]byte(v), partitionMeta)
+		if err != nil {
+			return nil, err
+		}
+		partitions = append(partitions, model.UnmarshalPartitionModel(partitionMeta))
+	}
+	return partitions, nil
+}
 
-	collMeta := &pb.CollectionInfo{}
-	err = proto.Unmarshal([]byte(collVal), collMeta)
+func fieldVersionAfter210(collMeta *pb.CollectionInfo) bool {
+	return len(collMeta.GetSchema().GetFields()) <= 0
+}
+
+func (kc *Catalog) listFieldsAfter210(ctx context.Context, collectionID typeutil.UniqueID, ts typeutil.Timestamp) ([]*model.Field, error) {
+	prefix := buildFieldPrefix(collectionID)
+	_, values, err := kc.Snapshot.LoadWithPrefix(prefix, ts)
+	if err != nil {
+		return nil, err
+	}
+	fields := make([]*model.Field, 0, len(values))
+	for _, v := range values {
+		partitionMeta := &schemapb.FieldSchema{}
+		err := proto.Unmarshal([]byte(v), partitionMeta)
+		if err != nil {
+			return nil, err
+		}
+		fields = append(fields, model.UnmarshalFieldModel(partitionMeta))
+	}
+	return fields, nil
+}
+
+func (kc *Catalog) GetCollectionByID(ctx context.Context, collectionID typeutil.UniqueID, ts typeutil.Timestamp) (*model.Collection, error) {
+	collKey := buildCollectionKey(collectionID)
+	collMeta, err := kc.loadCollection(ctx, collectionID, ts)
 	if err != nil {
 		log.Error("collection meta marshal fail", zap.String("key", collKey), zap.Error(err))
 		return nil, err
 	}
 
-	return model.UnmarshalCollectionModel(collMeta), nil
+	collection := model.UnmarshalCollectionModel(collMeta)
+
+	if partitionVersionAfter210(collMeta) {
```

Comment:
We' better  check whether need to process compatibility explicitly, it should return collection meta if the version is less than 210

Reply:
OK, fixed.